04/app.js

```diff
@@ -12,14 +12,22 @@ class App extends React.Component {
     }
 
     renderUsersList() {
-        const {users} = this.state;
-        return users.map(name => {
+        const {users, searchQuery} = this.state;
+        if(searchQuery === ''){
+            return users.map(name => {
             return (
                 <li onClick={ this.clickHandler }>
                     { name }
                 </li>
             );
         });
+        } else {
+            const filteredUsers = users.filter(users => user.includes(searchQuery));
```

Comment:
Warto sprowadzić wartości do tego samego "mianownika" czyli obie wartości zamienić na małe litery, aby nie miało to znaczenia przy wyszukiwaniu tj. `user.toLowerCase().includes(searchQuery.toLowerCase())`

PS. Zamiast `users` wew. `.filter` powinno być `user`